expression/expression.go

```diff
@@ -110,53 +110,52 @@ func (t ComparisonExpression) String() string {
 	return fmt.Sprintf("(%s %s %s)", l, t.Comparison, r)
 }
 
-/* Logical Node
-
-type LogicalNode struct {
-	Left    Node
-	Right   Node
+// Logical expressions
+// Contains expressions such as a Or b, a And b
+type LogicalExpression struct {
+	Left    Expression
+	Right   Expression
 	Logical Logical
-	Objects []Object
 }
 
-func (t LogicalNode) Evaluate(ctx Context) (interface{}, error) {
+func (t LogicalExpression) Evaluate(ctx Context) (interface{}, error) {
 	if t.Left == nil {
 		return nil, errors.New("Left node is nil")
 	}
+
 	if t.Right == nil {
 		return nil, errors.New("Right node is nil")
 	}
 
-	left, err := t.Left.Evaluate(tracer, ctx)
+	left, err := t.Left.Evaluate(ctx)
 	if err != nil {
-		return nil, errors.Wrapf(err, "Left logical evaluate failed")
+		return nil, errors.Wrap(err, "Left comparison evaluate failed")
 	}
 
-	right, err := t.Right.Evaluate(tracer, ctx)
+	right, err := t.Right.Evaluate(ctx)
 	if err != nil {
-		return nil, errors.Wrapf(err, "Right logical evaluate failed")
+		return nil, errors.Wrap(err, "Right comparison evaluate failed")
 	}
 
-	var result bool
+	if left == nil || right == nil {
+		return nil, nil
+
+	}
+
+	// Making this a switch statement so that we can add more operations later
 	switch t.Logical {
-	case And:
-		result, err = OperationAnd(left, right)
 	case Or:
-		result, err = OperationOr(left, right)
-	default:
-		result, err = false, LogicalError
+		return OperationOr(left, right)
 	}
```

Comment:
Add a default case that returns an error for unsupported logical expressions or LogicalError similar to ComparisionExpression